src/components/inputs/TextField.js

```diff
@@ -509,15 +510,17 @@ export default class TextField extends BaseInput {
   }
 
   renderRightButton() {
-    const {rightButtonProps} = this.getThemeProps();
-    const tintColor = _.get(rightButtonProps, 'style.tintColor', Colors.blue30);
-
     if (this.shouldDisplayRightButton()) {
+      const {rightButtonProps} = this.getThemeProps();
+      const tintColor = _.get(rightButtonProps, 'style.tintColor', Colors.blue30);
```

Comment:
this is kinda weird. 
first, accessing to style as an object is not so great, cause it can be a number or an array
second, tintColor is an image style prop, not a button. 
maybe to add an explicit prop to `rightButtonProps` for `tintColor` or `iconColor`

Reply:
Done. Thanks!